### ISSUE TITLE
Failed to connect to rabbitmq: Server connection reset:

### ISSUE THREAD

**Issue by Kadoronel** [its-a-feature/Mythic#370]
Je rencontre ce problème Failed to connect to rabbitmq: Server connection reset:  quand je consulte les logs après absence du callaback! j'ai besoin de vos orientations

**Comment by its-a-feature** [comment]
Can you provide a little more information about what you're seeing? 
* if you look at `sudo ./mythic-cli status` does it show everything as up and healthy? A good sign that something's wrong would be a container that is up for less than 30s while the other containers have been up for a while (reboot loop essentially)
* was this server running fine for a while, but now you're seeing an error? 
* You said you failed to get a callback and now see these errors - can you explain what happened a little more?

**Comment by Kadoronel** [comment]
after implementing the payload I do not have the return to the active callback level

**Comment by its-a-feature** [comment]
What operating system are you using? Kali?
Can you do a `sudo docker ps` and see if there are any other containers running? It kind of seems like the `mythic_rabbitmq` container and something else might be fighting over a port?
Can you do `sudo ./mythic-cli logs mythic_rabbitmq` and see the logs for that container?

**Comment by Kadoronel** [comment]
![Capture d’écran 2024-03-18 134419](https://github.com/its-a-feature/Mythic/assets/48646013/a323a225-ac9b-41b6-9bb1-45621b87895b)

**Comment by Kadoronel** [comment]
Linux kali 6.6.9-amd64 #1 SMP PREEMPT_DYNAMIC Kali 6.6.9-1kali1 (2024-01-08) x86_64 GNU/Linux

**Comment by Kadoronel** [comment]
![Capture d’écran 2024-03-18 135059](https://github.com/its-a-feature/Mythic/assets/48646013/165faab9-cfd3-4a8b-8ce7-a571dba85e08)

**Comment by Kadoronel** [comment]
logs mythic_rabbittmq

**Comment by its-a-feature** [comment]
Somebody else had a similar issue and it was due to a failed agent install. Are you able to re-install the agents and c2 you currently have installed? You can do this with `sudo ./mythic-cli install github [url] -f` (the `-f` will force it so you don't have to hit yes a few times

**Comment by Kadoronel** [comment]
I did this but the problem still persists and I can't get callback active

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/9cde3df7-0f48-46a9-aa6b-db6094c79d3a)

**Comment by its-a-feature** [comment]
oh ok, so it's showing apollo synced successfully which means it was able to connect to rabbitmq. Are you able to generate apollo payloads successfully?

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/7467604c-a7aa-416c-8afd-4667d0f8ff9f)

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/81a4ae31-802e-4bcf-b418-cf14c312ebc9)

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/1c0210f0-28c1-45ab-aee8-53c3e12b8afc)

**Comment by Kadoronel** [comment]
I can't get the machine's interact callback

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/201ed040-6920-4457-a66d-6ec29a1a1b08)

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/d1898004-43f9-40a9-9a97-c35995b8c21a)

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/426e065c-41f1-442a-8700-158d3bc95cdc)

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/919a9003-1b57-44e7-ada8-b1e6bfd99e29)

**Comment by its-a-feature** [comment]
Ah, i see you're connecting to a callback host via `https` and a callback port of `80`. Is that what you expect? By default, the `http` profile is listening via `http` on port `80`. Try going to the c2 profile page, clicking the dropdown for the http profile and clicking to edit the config. In there, there's an entry for use_ssl - set that to `true` then click `stop` and `start` in the UI for that profile. That should restart it to use `https` on port `80` like your config specifies. At that point you should get your callback

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/071bec31-0ecb-4b49-94e2-799fb12557bc)

**Comment by Kadoronel** [comment]
Thanks a lot

**Comment by its-a-feature** [comment]
awesome!! I wish we knew what was causing that weird server connection issue, but it seems like everything is working for you now?

**Comment by Kadoronel** [comment]
yes, I will continue and write to you about this

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/782eec0d-4e6e-41c1-b497-f8a02e3982f2)

**Comment by Kadoronel** [comment]
for downloading a file to the desktop of the machine with the path c:\users\bob\desktop I got this error back

**Comment by its-a-feature** [comment]
seems like it doesn't like something about the path. If you click to expand the task like you did in the screenshot, then the gray cog and select the keyboard icon ("View All Parameters"), what does that show?

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/df76489f-780e-49fb-a346-5c136a9bb7dc)

**Comment by Kadoronel** [comment]
I get the same error when I write it this way

**Comment by its-a-feature** [comment]
Oh, yeah it looks like it got all sorts of messed up and nested twice. How did you issue the task? 
Try doing one of the following:

- type "download" then press shift+enter to open up the modal. Type out the path here (just `C:\Users\bob\Desktop\filename`)
- type it out manually with `download -Path C:\Users\bob\Desktop\filename`

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/21473b36-24ea-4270-8f8d-96dcc66fe920)

**Comment by Kadoronel** [comment]
here

**Comment by its-a-feature** [comment]
yup, in that bottom one that says "Required", put the path there (C:\Users\bob\Desktop\filename)

**Comment by Kadoronel** [comment]
i still have the same errors

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/bfde317f-2c55-48fb-814c-256bd18fc694)

**Comment by its-a-feature** [comment]
Looks like there might be an issue catching an error, but the path also looks wrong. It says `C:\Users\bob\Desktopr\filename` (notice the extra r at the end of Desktop). Also, to help with troubleshooting on my end, are there any special characters in the path? like anything with a french accent mark?

**Comment by Kadoronel** [comment]
no french 
![image](https://github.com/its-a-feature/Mythic/assets/48646013/1dec25f1-737c-4676-97b1-a2c47e9804b4)

**Comment by Kadoronel** [comment]
I have capital letters and i's in them, that shouldn't cause any problems

**Comment by Kadoronel** [comment]
if I'm not mistaken

**Comment by its-a-feature** [comment]
Now it looks like you're getting an access denied error 🤔 does your user have access to that file?

**Comment by Kadoronel** [comment]
yes my user mode has access to files, better on the command line I can do ls and cat but I can't do downloads

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/78bd703b-7637-4d5a-a661-9c6077837b52)

**Comment by Kadoronel** [comment]
reg_write_value "HKEY_CURRENT_USER\Control Panel\Desktop\Colors" "MenuText" "255 255 255"

**Comment by Kadoronel** [comment]
Unhandled exception: System.NullReferenceException: La référence d'objet n'est pas définie à une instance d'un objet.

   à Tasks.reg_write_value.SetValue(String hive, String subkey, String valueName, Object valueValue)

   à Tasks.reg_write_value.Start()

   à ApolloInterop.Classes.Tasking.<CreateTasking>b__7_0()

   à System.Threading.Tasks.Task.Execute()

**Comment by DragoQCC** [comment]
Hey @Kadoronel I'm looking into this error can you share the file permissions on the file where cat works but not download? As well as the full commands ran with the file name, path, and file extension if you don't mind sharing those? Currently as long as read permissions are enabled I can read & download files fine or it without it will fail.  I'm just trying to reproduce this error on my test system so I want my setup to match as best I can.

**Comment by Kadoronel** [comment]
![image](https://github.com/its-a-feature/Mythic/assets/48646013/43f1db5a-a15f-498a-81ce-39e943b1ab97)

**Comment by DragoQCC** [comment]
seems like a permission deny if you do a ls on the parent folder so i think that would be `ls c:\users\administrateur` then under actions I think there should be a acl / view permissions command that can be issued so we know if the user running the implant can read the folder. Also just in case when doing a download it must be to a specific file and not a whole directory.

**Comment by Kadoronel** [comment]
Thank you for your feedback on the question, I achieved this differently without using the command lines. I don't understand why this doesn't work well on the command line. I continue to research on the subject

**Comment by its-a-feature** [comment]
Were you able to figure out what's going on and why you had that issue?

**Comment by its-a-feature** [comment]
Since this thread hasn't had any traffic for a while, I'm going to close this issue, but please feel free to open it back up or open a new one if you have more issues in the future :)

**Comment by Harsh595fy** [comment]
i got this error 

![Image](https://github.com/user-attachments/assets/25d4efcc-2821-494d-a664-45e94bd4a899)

**Comment by its-a-feature** [comment]
@Harsh595fy - that means that your browser lost its connection to the Mythic server's websocket connections. A page refresh should bring it back. I typically see this after a period of time when a browser requires you to accept the self-signed certificate again
